svc-update/update/simpleUpdate.go

```diff
@@ -19,12 +19,128 @@ package update
+		log.Println(errMsg)
+		return common.GeneralError(http.StatusInternalServerError, response.InternalError, errMsg, nil, nil)
+	} else if invalidProperties != "" {
+		errorMessage := "error: one or more properties given in the request body are not valid, ensure properties are listed in uppercamelcase "
+		log.Println(errorMessage)
+		response := common.GeneralError(http.StatusBadRequest, response.PropertyUnknown, errorMessage, []interface{}{invalidProperties}, nil)
+		return response
+	}
+	for _, individualTarget := range updateRequest.Targets {
+		// spliting the uuid and system id
+		requestData := strings.Split(individualTarget, "/")
+		var requestTarget []string
+		for _, data := range requestData {
+			if strings.Contains(data, ":") {
+				requestTarget = strings.Split(data, ":")
+			}
+		}
+		if len(requestTarget) != 2 || requestTarget[1] == "" {
+			errorMessage := "error: SystemUUID not found"
+			return common.GeneralError(http.StatusBadRequest, response.ResourceNotFound, errorMessage, []interface{}{"System", individualTarget}, nil)
+		}
+		uuid := requestTarget[0]
+		target, gerr := e.External.GetTarget(uuid)
+		if gerr != nil {
+			return common.GeneralError(http.StatusBadRequest, response.ResourceNotFound, gerr.Error(), []interface{}{"System", uuid}, nil)
+		}
+
+		decryptedPasswordByte, err := e.External.DevicePassword(target.Password)
+		if err != nil {
+			// Frame the RPC response body and response Header below
+			errorMessage := "error while trying to decrypt device password: " + err.Error()
+			return common.GeneralError(http.StatusInternalServerError, response.InternalError, errorMessage, nil, nil)
+		}
+		target.Password = decryptedPasswordByte
+
+		// Get the Plugin info
+		plugin, gerr := e.External.GetPluginData(target.PluginID)
+		if gerr != nil {
+			errorMessage := "error while trying to get plugin details"
+			return common.GeneralError(http.StatusInternalServerError, response.InternalError, errorMessage, nil, nil)
+		}
+		var contactRequest ucommon.PluginContactRequest
+		contactRequest.ContactClient = e.External.ContactClient
+		contactRequest.Plugin = plugin
+
+		if strings.EqualFold(plugin.PreferredAuthType, "XAuthToken") {
+			var err error
+			contactRequest.HTTPMethodType = http.MethodPost
+			contactRequest.DeviceInfo = map[string]interface{}{
+				"UserName": plugin.Username,
+				"Password": string(plugin.Password),
+			}
+			contactRequest.OID = "/ODIM/v1/Sessions"
+			_, token, getResponse, err := e.External.ContactPlugin(contactRequest, "error while creating session with the plugin: ")
+
+			if err != nil {
+				return common.GeneralError(getResponse.StatusCode, getResponse.StatusMessage, err.Error(), nil, nil)
+			}
+			contactRequest.Token = token
+		} else {
+			contactRequest.BasicAuth = map[string]string{
+				"UserName": plugin.Username,
+				"Password": string(plugin.Password),
+			}
+
+		}
+		replaceTargetUUID := strings.Replace(string(req.RequestBody), uuid+":", "", -1)
```

Comment:
the request which is sent to plugin needs to /ODIM  resource links instead of /redfish/ please update the same